api_handlers.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Google Inc.
```

Comment:
I am wholly unqualified to review Go code (at least, right now). @jeffcarp can you plz look at that bit of this PR?